src/reducers/secrets.js

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { combineReducers } from 'redux';
+import keyBy from 'lodash.keyby';
+
+function byId(state = {}, action) {
+  switch (action.type) {
+    case 'SECRETS_FETCH_SUCCESS':
+      return keyBy(action.data, 'name');
+    case 'SECRET_DELETE_SUCCESS':
+      const newState = state;
+      delete newState[action.id];
+      return newState;
+    default:
+      return state;
+  }
+}
+
+function byNamespace(state = {}, action) {
+  switch (action.type) {
+    case 'SECRETS_FETCH_SUCCESS':
+      const secrets = [];
+      action.data.forEach((secret, index) => {
+        const object = {
+          uid: index.toString(),
+          name: secret.name,
+          annotations: secret.url
+        };
+        secrets.push(object);
+      });
+
+      const { namespace } = action;
+      return {
+        ...state,
+        [namespace]: secrets
+      };
+    default:
+      return state;
+  }
+}
+
+function isFetching(state = false, action) {
+  switch (action.type) {
+    case 'SECRETS_FETCH_REQUEST':
+    case 'SECRET_DELETE_REQUEST':
```

Comment:
Is there a plan to use these to render some loading / progress state in the UI? If so, do we need to differentiate between them or would there be a common loading state for all? With this implementation it wouldn't be possible to differentiate between an in-progress delete or a request to load all secrets for example.

Reply:
Right now its a common loading state that renders the table skeleton in the UI. As for now, when the user deletes the secrets it automatically makes a requests to load all the secrets so I don't see a reason to differentiate the loading state.